x-pack/plugins/observability_solution/apm/public/components/app/service_logs/index.tsx

```diff
@@ -62,17 +76,30 @@ export function ClassicServiceLogsStream() {
     [environment, kuery, serviceName, start, end]
   );
 
-  return (
-    <LogStream
-      logView={{ type: 'log-view-reference', logViewId: 'default' }}
-      columns={[{ type: 'timestamp' }, { type: 'message' }]}
+  const logSources = useAsync(logSourcesService.getFlattenedLogSources);
+
+  const dependencies = useMemo(() => {
+    return { embeddable, savedSearch, searchSource, dataViews };
+  }, [dataViews, embeddable, savedSearch, searchSource]);
+
+  const timeRange = useMemo(() => ({ from: start, to: end }), [start, end]);
```

Comment:
I see `query` is passed inline and not memoized, which would retrigger the effect in SavedSearchComponent that sets the initial serialized state even if these `dependencies` and `timeRange` are behind useMemo.
Same applies for `displayOptions`, is there a reason why some of these are memoized and others are not? Also, do we need memoization here since we can look at individual properties in the dependency array of the SavedSearchComponent side effect?